### PR TITLE
add isHeadless capability on docs

### PR DESCRIPTION
**Pull request by vrunoa**
## Proposed changes

Update documentation with the new isHeadless capability available for ios and android

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/en/writing-running-appium/caps.md [hunk 1]
```diff
@@ -69,6 +69,7 @@
 |`autoGrantPermissions`|Have Appium automatically determine which permissions your app requires and grant them to the app on install. Defaults to `false`|`true` or `false`|
 |`networkSpeed`|Set the network speed emulation. Specify the maximum network upload and download speeds. Defaults to `full`| `['full','gsm', 'edge', 'hscsd', 'gprs', 'umts', 'hsdpa', 'lte', 'evdo']` Check [-netspeed option](https://developer.android.com/studio/run/emulator-commandline.html) more info about speed emulation for avds|
 |`gpsEnabled`|Toggle gps location provider for emulators before starting the session. By default the emulator will have this option enabled or not according to how it has been provisioned.|`true` or `false`|
+|`isHeadless`|Set this capability to `true` to run the Emulator headless when device display is not needed to be visible. `false` is the default value.|e.g., `true`|
```
Comment (imurchie) [inline docs/en/writing-running-appium/caps.md]: Can you add a note to see the iOS capabilities for more details about that system?

File: docs/en/writing-running-appium/caps.md [hunk 2]
```diff
@@ -102,7 +103,7 @@
 |`appName`| The display name of the application under test. Used to automate backgrounding the app in iOS 9+.|e.g., `UICatalog`|
 |`customSSLCert`|(Sim/Emu-only) Add an SSL certificate to simulator. | e.g. <br/>`-----BEGIN CERTIFICATE-----MIIFWjCCBEKg...`<br/>`-----END CERTIFICATE-----`|
 |`webkitResponseTimeout`|(Real device only) Set the time, in ms, to wait for a response from WebKit in a Safari session. Defaults to `5000`|e.g., `10000`|
-
+|`isHeadless`|Set this capability to `true` if automated tests are running on Simulator and the device display is not needed to be visible. This only has an effect since Xcode9 and only for simulators. All running instances of Simulator UI are going to be automatically terminated if headless test is started. `false` is the default value.|e.g., `true`|
```
Comment (imurchie) [inline docs/en/writing-running-appium/caps.md]: This is technically an "iOS Only, using XCUITest" capability.

**Review by imurchie — CHANGES_REQUESTED** [review]

File: docs/en/writing-running-appium/caps.md (hunk 1 above)
Comment (vrunoa) [inline docs/en/writing-running-appium/caps.md]: What should I add ? Sorry, I dont finish to understand what am I missing.

File: docs/en/writing-running-appium/caps.md [hunk 3]
```diff
@@ -103,10 +104,10 @@
 |`customSSLCert`|(Sim/Emu-only) Add an SSL certificate to simulator. | e.g. <br/>`-----BEGIN CERTIFICATE-----MIIFWjCCBEKg...`<br/>`-----END CERTIFICATE-----`|
 |`webkitResponseTimeout`|(Real device only) Set the time, in ms, to wait for a response from WebKit in a Safari session. Defaults to `5000`|e.g., `10000`|
 
-
 ### iOS Only, using XCUITest
 
 (For XCUITest-specific capabilities, please refer to the documentation on the [XCUITest driver](https://github.com/appium/appium-xcuitest-driver#desired-capabilities) itself.)
+|`isHeadless`|Set this capability to `true` if automated tests are running on Simulator and the device display is not needed to be visible. This only has an effect since Xcode9 and only for simulators. All running instances of Simulator UI are going to be automatically terminated if headless test is started. `false` is the default value.|e.g., `true`|
```
Comment (SrinivasanTarget) [inline docs/en/writing-running-appium/caps.md]: @vrunoa Do we really need to duplicate?

Comment (SrinivasanTarget) [inline docs/en/writing-running-appium/caps.md]: Can we remove it here?

**Review by SrinivasanTarget — CHANGES_REQUESTED** [review]

File: docs/en/writing-running-appium/caps.md (hunk 3 above)
Comment (vrunoa) [inline docs/en/writing-running-appium/caps.md]: This is only available for ios xcuitest and android, not for the other platforms that appium supports, like win desktop or you.i tv. Don't know if they should be under general capabilities.

Comment (SrinivasanTarget) [inline docs/en/writing-running-appium/caps.md]: AFAIK If it is specific to XCUI we update it only [here](https://github.com/appium/appium-xcuitest-driver#desired-capabilities). Does `UIAutomator2` mode supports this caps too?

Comment (vrunoa) [inline docs/en/writing-running-appium/caps.md]: mmm I'm lost @SrinivasanTarget , haha. I think @imurchie ask me to move this to ios xcui section. Maybe I misunderstand him.

Comment (vrunoa) [inline docs/en/writing-running-appium/caps.md]: @SrinivasanTarget for android this an avd argument, should work not matter the automationName.

Comment (SrinivasanTarget) [inline docs/en/writing-running-appium/caps.md]: >should we point this is emulators only capability ?

Yes please.

Comment (vrunoa) [inline docs/en/writing-running-appium/caps.md]: Oh got it.

Comment (vrunoa) [inline docs/en/writing-running-appium/caps.md]: Actually on the descriptin it says: `to run the Emulator headless` so I think we dont need to add any more description.

Comment (SrinivasanTarget) [inline docs/en/writing-running-appium/caps.md]: Ok

**Review by SrinivasanTarget — APPROVED** [review]

File: docs/en/writing-running-appium/caps.md (hunk 1 above)
Comment (jlipps) [inline docs/en/writing-running-appium/caps.md]: @vrunoa maybe something like: `isHeadless is also supported on iOS, see below`

**Review by jlipps — APPROVED** [review]

**Review by imurchie — APPROVED** [review]

**Pull request merged**